src/command_modules/azure-cli-advisor/azure/cli/command_modules/advisor/custom.py

```diff
@@ -0,0 +1,143 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import time
+import uuid
+
+from azure.mgmt.advisor.models import ConfigData, ConfigDataProperties
+
+
+def cli_advisor_generate_recommendations(client, timeout=30):
+    response = client.generate(raw=True)
+    location = response.headers['Location']
+
+    operation_id = _parse_operation_id(location)
+
+    elapsedTime = 0
+    maxTime = int(timeout)
+    if maxTime < 1:
+        maxTime = 1
+    while elapsedTime < maxTime:
+        response = client.get_generate_status(
+            raw=True,
+            operation_id=operation_id
+        )
+        status_code = response.response.status_code
+        if status_code == 204:
+            return {'Status': 204, 'Message': 'Recommendation generation has been completed.'}
+        time.sleep(1)
+        elapsedTime += 1
+    return {'Status': 202, 'Message': 'Recommendation generation is in progress.'}
+
+
+def cli_advisor_list_recommendations(client, ids=None, rg_name=None, category=None):
+    scope = None
+    if ids:
+        for id_arg in ids:
+            subScope = 'ResourceId eq \'{}\''.format(id_arg)
+            if scope:
+                scope += 'and {}'.format(subScope)
+            else:
+                scope = subScope
+    elif rg_name:
+        scope = 'ResourceGroup eq \'{}\''.format(rg_name)
+    if category:
+        subScope = 'Category eq \'{}\''.format(category)
+        if scope:
+            scope += 'and {}'.format(subScope)
+        else:
+            scope = subScope
```

Comment:
Actually, you can define scope as a list then append all the parts to the list.
At the end, `'and '.join(scope)` and pass that to `client.list`.
Also, I'd recommend using the keyword argument for the call to `client.list`. Something like `client.list(filter=scope)`